### PR TITLE
Feature/gh 746 create2 property set

### LINKED ISSUE #746: Implement v2 of safe creation API
**Endpoints**

    POST /v2/safes
    GET /v2/safes/{address}/funded
    PUT /v2/safes/{address}/funded

**Reference**
- [DefaultTransactionExecutionRepository.kt][hash] - hash changed
- [CreateSafeConfirmRecoveryPhraseViewModel.kt][impl] - response assertions
- [CreateSafeConfirmRecoveryPhraseViewModelTest.kt][test] - test cases
- [Staging API][api] - api doc

[hash]: https://github.com/gnosis/safe-android/blob/78f5b83a02313ba7792b474b567f389b0ddfa9c8/app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultTransactionExecutionRepository.kt#L110
[impl]: https://github.com/gnosis/safe-android/blob/78f5b83a02313ba7792b474b567f389b0ddfa9c8/app/src/main/java/pm/gnosis/heimdall/ui/safe/create/CreateSafeConfirmRecoveryPhraseViewModel.kt
[test]: https://github.com/gnosis/safe-android/blob/78f5b83a02313ba7792b474b567f389b0ddfa9c8/app/src/test/java/pm/gnosis/heimdall/ui/safe/create/CreateSafeConfirmRecoveryPhraseViewModelTest.kt
[api]: https://safe-relay.staging.gnosisdev.com

### PR DESCRIPTION
**Pull request by DmitryBespalov**
closes #746 

Note: you'll have to update the AppConfig.yml on your machine from encrypted files after this is merged.

### CODE REVIEW

File: MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift [hunk 1]
```diff
@@ -71,14 +72,19 @@ public class DeploymentDomainService {
...
         handleError { wallet in
-            let s = DomainRegistry.encryptionService.ecdsaRandomS()
-            let request = SafeCreationTransactionRequest(owners: wallet.allOwners().map { $0.address },
-                                                         confirmationCount: wallet.confirmationCount,
-                                                         ecdsaRandomS: s)
+            let owners = wallet.allOwners().map { $0.address }
+            let request = SafeCreationRequest(saltNonce: DomainRegistry.encryptionService.randomSaltNonce(),
+                                              owners: owners,
+                                              confirmationCount: wallet.confirmationCount,
+                                              paymentToken: .zero)
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift]: .zero sounds weird. How about .Ether ?

File: MultisigWallet/MultisigWalletDomainModel/Wallet/MockTransactionRelayService1.swift [hunk 2]
```diff
@@ -8,21 +8,20 @@ import XCTest
 class MockTransactionRelayService1: TransactionRelayDomainService {
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet/MockTransactionRelayService1.swift:8]: Should the name end with "1"?

File: MultisigWallet/MultisigWalletDomainModel/Wallet/RecoveryDomainService.swift [hunk 3]
```diff
@@ -116,13 +100,21 @@ public class RecoveryDomainService: Assertable {
 
     private func pullWalletData() throws {
         let wallet = DomainRegistry.walletRepository.selectedWallet()!
-        let contract = SafeOwnerManagerContractProxy(wallet.address!)
-        let existingOwnerAddresses = try contract.getOwners()
-        let confirmationCount = try contract.getThreshold()
+        let ownerContract = SafeOwnerManagerContractProxy(wallet.address!)
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet/RecoveryDomainService.swift]: confusing. Looks like the ownerContranct bounded to the wallet address.

File: MultisigWallet/MultisigWalletDomainModel/Wallet/SafeCreationResponseValidator.swift [hunk 4]
```diff
@@ -9,51 +9,72 @@ import BigInt
...
-    var isValid: Bool {
-        guard let v = Int(v) else { return false }
-        return ECDSASignatureBounds.isWithinBounds(r: r, s: s, v: v)
+    public func validate(_ response: SafeCreationRequest.Response, request: SafeCreationRequest) throws {
+        try assertEqual(response.paymentToken, request.paymentToken, SafeCreationValidationError.invalidPaymentToken)
+        try assertTrue(repository.isValidMasterCopy(address: response.masterCopyAddress),
+                       SafeCreationValidationError.invalidMasterCopy)
+        try assertTrue(repository.isValidProxyFactory(address: response.proxyFactoryAddress),
+                       SafeCreationValidationError.invalidProxyFactory)
+        try assertTrue(repository.isValidPaymentRecevier(address: response.paymentReceiverAddress),
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet/SafeCreationResponseValidator.swift:36]: Is it a relayer account address?

File: MultisigWallet/MultisigWalletDomainModel/Wallet/WalletSettingsDomainService.swift [hunk 5]
```diff
@@ -51,8 +48,8 @@ public class WalletSettingsDomainService {
                                                  old: oldOwner2,
                                                  new: fakeNewOwner2)
                 ownerList.replace(oldOwner2, with: fakeNewOwner2)
-
-                let multiSendProxy = MultiSendContractProxy(config.multiSendContractAddress)
+                let multiSendAddess = DomainRegistry.safeContractMetadataRepository.multiSendContractAddress
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet/WalletSettingsDomainService.swift]: typo "Addess"

File: MultisigWallet/MultisigWalletDomainModel/Wallet/SafeCreationResponseValidator.swift (hunk 4 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Wallet/SafeCreationResponseValidator.swift:36]: this is a 'funder' address

File: MultisigWallet/MultisigWalletDomainModel/Wallet/MockTransactionRelayService1.swift (hunk 2 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Wallet/MockTransactionRelayService1.swift:8]: yeah.. there is another mock without '1'

File: MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift (hunk 1 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Wallet/DeploymentDomainService.swift]: yes, better

File: MultisigWallet/MultisigWalletDomainModel/Wallet/WalletSettingsDomainService.swift (hunk 5 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Wallet/WalletSettingsDomainService.swift]: ok

File: SafeUIKit/SafeAppUI/AppFlow/MainFlow/Send/SendInputViewModel.swift [hunk 6]
```diff
@@ -33,7 +33,7 @@ class SendInputViewModel {
     private var walletService: WalletApplicationService { return ApplicationServiceRegistry.walletService }
     private let updateBlock: () -> Void
 
-    init(tokenID: BaseID, onUpdate: @escaping () -> Void) {
```
Comment (sche) [inline SafeUIKit/SafeAppUI/AppFlow/MainFlow/Send/SendInputViewModel.swift:36]: please add comment

File: safe/safe/IntegrationTests/Safe.swift [hunk 7]
```diff
@@ -15,13 +15,45 @@ struct Safe {
...
+        walletID = wallet.id
+    }
 
     var _test: GnosisTransactionRelayServiceTests!
 
     var gasAdjustment: BigInt = 0
 
     var proxy: SafeOwnerManagerContractProxy { return SafeOwnerManagerContractProxy(address) }
 
+
```
Comment (sche) [inline safe/safe/IntegrationTests/Safe.swift]: extra space

**Review by sche — APPROVED** [review]
👍

**Pull request merged**